solidity/contracts/converter/BancorConverter.sol

```diff
@@ -977,7 +998,13 @@ contract BancorConverter is IBancorConverter, TokenHandler, SmartTokenController
         return _supply.mul(_amount).mul(reserveRatio).div(_balance.add(_amount).mul(WEIGHT_RESOLUTION));
     }
 
-    function ceilLog(uint256 _x) public pure returns (uint256) {
+    /**
+      * @dev calculates the number of decimal digits in a given value
+      * 
+      * @param _x   value (assumed positive)
+      * @return the number of decimal digits in the given value
+    */
+    function decimalLength(uint256 _x) public pure returns (uint256) {
         uint256 y = 0;
         while (_x > 0) {
             _x /= 10;

```

Comment:
getting a warning here - assigning to function argument.
also, can just do y++ instead of y += 1

Reply:
@yudilevi:
1. With `npm run build` or `truffle compile`? Weird, I don't get that. In any case, we can either suppress that warning in your compiler settings, or declare something like `uint256 z = _x` at the beginning, and then use `z /= 10`.
2. We can; I typically prefer to avoid using `++` outside of `for` loops (and in this case I chose the `+= 1` also because of the `/= 10`).

Reply:
1. I think newer compilers do that and that's why I'm getting in
2. ...

Reply:
@yudilevi: How is it possible that you're on a newer compiler. We should both be on solc 0.4.26, and if we make sure to use our Truffle version (4.1.16), then that **will** be the case.

Reply:
I checked, it's a linter error: "security/no-assign-params: "decimalLength": Avoid assigning to function parameters."

Reply:
@yudilevi: So maybe change that linter-rule, because there is nothing wrong with assigning a function parameter.

Reply:
we'll leave it for now but eventually we'll want to comply with industry standards and conventions.

Reply:
@yudilevi : As far as I remember, this actually used to be the standard in an older version of solc (0.4.18 or similar), and was later removed. So if anything, the industry standard is to allow changing input arguments.

Reply:
if that was the standard, solium linter wouldn't have pointed it out as a security issue but like I said, we can leave it there for now so feel free to resolve this one.

Reply:
@yudilevi: Fixed.